pkg/controller/operands/cliDownload.go

```diff
@@ -1,57 +1,72 @@
 
 	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1beta1"
 	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/controller/common"
 	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
 	consolev1 "github.com/openshift/api/console/v1"
-	objectreferencesv1 "github.com/openshift/custom-resource-status/objectreferences/v1"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/tools/reference"
 )
 
-type CLIDownloadHandler genericOperand
+type cLIDownloadHandler genericOperand
 
-func (h CLIDownloadHandler) Ensure(req *common.HcoRequest) error {
-	ccd := NewConsoleCLIDownload(req.Instance)
-
-	found := NewConsoleCLIDownload(req.Instance)
-	err := hcoutil.EnsureCreated(req.Ctx, h.Client, found, req.Logger)
-	if err != nil {
-		if meta.IsNoMatchError(err) {
-			req.Logger.Info("ConsoleCLIDownload was not found, skipping")
-		}
-		return err
+func newCLIDownloadHandler(Client client.Client, Scheme *runtime.Scheme) *cLIDownloadHandler {
+	return &cLIDownloadHandler{
+		Client:                 Client,
+		Scheme:                 Scheme,
+		crType:                 "ConsoleCLIDownload",
+		removeExistingOwner:    false,
+		setControllerReference: false,
+		hooks:                  &cLIDownloadHooks{},
 	}
+}
 
-	// Make sure we hold the right link spec
-	if reflect.DeepEqual(found.Spec, ccd.Spec) {
-		objectRef, err := reference.GetReference(h.Scheme, found)
-		if err != nil {
-			req.Logger.Error(err, "failed getting object reference for ConsoleCLIDownload")
-			return err
+type cLIDownloadHooks struct{}
+
+func (h cLIDownloadHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	return NewConsoleCLIDownload(hc), nil
+}
+
+func (h cLIDownloadHooks) getEmptyCr() client.Object {
+	return &consolev1.ConsoleCLIDownload{}
+}
+
+func (h cLIDownloadHooks) postFound(_ *common.HcoRequest, _ runtime.Object) error { return nil }
+
+func (h cLIDownloadHooks) getObjectMeta(cr runtime.Object) *metav1.ObjectMeta {
+	return &cr.(*consolev1.ConsoleCLIDownload).ObjectMeta
+}
+
+func (h *cLIDownloadHooks) updateCr(req *common.HcoRequest, Client client.Client, exists runtime.Object, required runtime.Object) (bool, bool, error) {
+	desired, ok1 := required.(*consolev1.ConsoleCLIDownload)
+	found, ok2 := exists.(*consolev1.ConsoleCLIDownload)
+	if !ok1 || !ok2 {
+		return false, false, errors.New("can't convert to ConsoleCLIDownload")
+	}
+	if !reflect.DeepEqual(found.Spec, desired.Spec) ||
+		!reflect.DeepEqual(found.Labels, desired.Labels) {
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing ConsoleCLIDownload's Spec to new opinionated values")
+		} else {
+			req.Logger.Info("Reconciling an externally updated ConsoleCLIDownload's Spec to its opinionated values")
 		}
-		err = objectreferencesv1.SetObjectReference(&req.Instance.Status.RelatedObjects, *objectRef)
```

Comment:
How do you handle this?

Reply:
I thought: We have the similar functionality here: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/pkg/controller/operands/operand.go#L145 and it is a part of operandHandler. Therefore, we don't need to have such  a logic here. Am I wrong?

Reply:
No, you are right - I was confused with https://github.com/kubevirt/hyperconverged-cluster-operator/blob/master/pkg/controller/operands/operand.go#L160

Reply:
Thanks!